### ISSUE TITLE
Missing Project Name on Components Page

### ISSUE THREAD

**Issue by msymons** [DependencyTrack/dependency-track#1166]
### Current Behavior:

When a search is performed on the Components page, the results (by default) include a column "Project Name".

The Project Name is mostly populated but sometimes it is blank...

![image](https://user-images.githubusercontent.com/4938718/130460440-9946ccb1-a2bb-4adb-958a-9c413fdc241a.png)

Clicking through to see component details confirms that there is a valid project.

This behaviour was first noticed when v4.0 was in Beta but still exists in v4.3.3.

### Steps to Reproduce:

* Navigate in UI to Components Page
* Perform a search that you know will return a lot of results.  The screenshot above was for name = `jackson-databind`, but just `jackson` would give many more.   Or search for group = `apache`.  All this is based on the assumption that DT has many maven projects!
*  Change page display to 100 rows.
*  Inspect the "Project Name" column,  checking for missing entries
*  If nothing is blank then navigate to page 2,3,4, etc.

### Expected Behavior:

The should be no blank project names.

### Environment:
  
  - Dependency-Track Version: 4.3.3
  - Distribution: [ Docker ]
  - BOM Format & Version: CycloneDX 1.2 schema
  - Database Server: [ PostgreSQL ]
  - Browser: Firefox, Chrome.

**Comment by hgalvao** [comment]
Exactly same problem here.
Dependency-Track API Server Version: 4.3.6
Dependency-Track Frontend Version: 4.3.1
Response from server in a good row: `project: {name: "Project Name", ...}`
Response from server in a missing row: `project: {active: true}`

**Comment by msymons** [comment]
I have performed testing using my test-server (using snapshot build) and things now do seem to be working a bit better.. but there are definitely still times when project name does not display when performing component search.  

There IS a simple work-around,  which is good for 4.3.6 or 4.4.0-SNAPSHOT
* CTRL-CLICK on component link in any "problem row".  This opens a new tab for the component (which you can immediately close).
* Click the components screen refresh button in the UI (not the browser refresh button!).
* All missing project names (link) now display.

```
Dependency-Track v4.4.0-SNAPSHOT
Build ID: 30afa4da-6da1-4482-b3da-d15ce994b12e
Built On: 14 Dec 2021 at 23:14:54

Frontend v4.3.1
Build ID: ca8eaf30-0656-40f1-8ee8-c3375d576a8b
Built On: 7 Dec 2021 at 18:06:17 
```

**Comment by msymons** [comment]
This has been fixed in v4.4.0 via commits linked to #1227.
